common/cache.py

```
@@ -0,0 +1,190 @@
+        else:
+            raise AttributeError("View must have a 'primary_model' attribute.")
+
+        # Add the cache_models names
+        cache_models = getattr(self, "cache_models", [])
+        model_names.extend(model.__name__ for model in cache_models)
+        # Combine the model names into a string
+        model_names_str = "_".join(model_names)
+
+        return f"{primary_model.__name__}:{self.__class__.__name__}_{model_names_str}_{user_id}_{query_params_hash}_cache_key"
+
+    def get_cached_response(self, cache_key) -> Union[Response | None]:
```

Ruff (0.12.2)

Missing type annotation for function argument `cache_key`

(ANN001)